pkg/parquetquery/iters.go

```diff
@@ -451,6 +453,11 @@ func (c *ColumnIterator) Next() (*IteratorResult, error) {
 }
 
 func (c *ColumnIterator) next() (RowNumber, pq.Value, error) {
+	if !c.init {
```

Comment:
the line below `c.currErr.Load()` seems to imply that we this struct may be accessed concurrently by multiple goroutines. i suppose we should either put similar protections here or drop synchronous access to `currErr`?

Reply:
The concurrent access is between the iterate goroutine and the thing calling Next() or SeekTo(). Only the couple things shared between them are protected.   `init` isn't read by the iterate goroutine so it's ok.